exec/java-exec/src/test/java/org/apache/drill/exec/record/TestMaterializedField.java

```diff
@@ -58,32 +58,24 @@ public void initialize() {
   }
 
   @Test
-  public void testClone() {
-    final MaterializedField cloneParent = parent.clone();
-    final boolean isParentEqual = parent.equals(cloneParent);
-    assertTrue("Cloned parent does not match the original", isParentEqual);
+  public void testCopy() {
+    final MaterializedField cloneParent = parent.copy();
+    assertEquals("Parent copy does not match the original", parent, cloneParent);
```

Comment:
Call me paranoid, but this is using one of the functions I'd be worried about. If someone omits some fields from the `equals()` implementation, we'll get a false positive.

Would seem better to do a full test of `equals()`, and only then use it in tests.

Reply:
The goal of the PR is to address usage of the `clone()` instead of the copy constructor with a few minor fixes of badly written Java code. The change here addresses the issue of using `assertTrue` where it should assert for equality.